File: Python/171.Excel Sheet Column Number.py
#把字符串转化成对应的数字，A->1，B->2....AA->27，AB->28
class Solution(object):
    def titleToNumber(self, s):
        s = s[::-1]
        result = ord(s[0])-64
        for i in range(1,len(s)):
            result +=  26**i*(ord(s[i])-64)
        return result


s = Solution()
print(s.titleToNumber("ABC"))

File: C++/Math/171.Excel Sheet Column Number.cpp
/*
字符串转换成整数。
	A -> 1
    B -> 2
    C -> 3
    ...
    Z -> 26
    AA -> 27
    AB -> 28 
*/

#include <iostream>
#include <cstring>
using namespace std;

class Solution {
public:
	int titleToNumber(string s) {
        int result = 0;
        for(char ch:s)
        {
            result = result*26+(ch-'A'+1);
        }
        return result;
    }
};
int main()
{
	Solution s;
	cout<<s.titleToNumber("ABC")<<endl;
}

